service/controller/v24/resource/stackoutput/create.go

```diff
@@ -48,6 +50,35 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		r.logger.LogCtx(ctx, "level", "debug", "message", "found the tenant cluster cloud formation stack outputs")
 	}
 
+	if r.route53Enabled {
+		v, err := cc.CloudFormation.GetOutputValue(outputs, key.HostedZoneNameServers)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		cc.Status.Cluster.HostedZoneNameServers = v
+	}
+
+	{
+		v, err := cc.CloudFormation.GetOutputValue(outputs, key.VPCPeeringConnectionIDKey)
+		if cf.IsOutputNotFound(err) {
+			// TODO this exception is necessary for clusters upgrading from v23 to
+			// v24. The code can be cleaned up in v25 and the controller context value
+			// assignment can be managed like the other examples below.
+			//
+			//     https://github.com/giantswarm/giantswarm/issues/5496
+			//
+			v, err := r.searchPeeringConnectionID(key.ClusterID(cr))
+			if err != nil {
+				return microerror.Mask(err)
+			}
+			cc.Status.Cluster.VPCPeeringConnectionID = v
```

Comment:
I figured out there was a deadlock during updates. The peering ID is not there in older clusters and when we update it is missing. We need it to update though, so the update got stuck. For the time being I added the fallback lookup which we basically had in the adapter.

Reply:
Should it be tracked to be removed?

Reply:
It already is.